src/server/plugin/Config.ts

```diff
@@ -40,12 +41,20 @@ export interface Config extends VerdaccioConfig {
 // Access
 //
 
-export function getConfig(config: Config, key: PluginConfigKey): string {
+export function getConfig<K extends PluginConfigKey>(
+  config: Config,
+  key: K,
+): PluginConfig[K] {
   const value =
-    get(config, `middlewares[${pluginName}][${key}]`) ??
-    get(config, `auth[${pluginName}][${key}]`)
-
-  return process.env[value] || value
+    get(config, ["middlewares", pluginName, key]) ??
+    get(config, ["auth", pluginName, key])
+
+  if (typeof value === "string") {
+    // @ts-ignore
+    return process.env[value] || value
```

Comment:
I updated the type of this function to be more correct but the feature of getting the value from the environment variable doesn't exactly play well with non-string types so this is still unsound. I could also just make `repository-access` be a string instead of just boolean.